### ISSUE TITLE
KaSim compile error: yojson 1.6 ?

### ISSUE THREAD

**Issue by hmedina** [Kappa-Dev/KaSim#551]
It seems updating Yojson breaks compilation of the kappa suite. Below is the output of an upgrade attempt.
```
$ opam upgrade
The following actions will be performed:
  ↗ upgrade   yojson         1.5.0 to 1.6.0
  ↻ recompile kappa-library  4~dev*         [uses yojson]
  ↻ recompile kappa-binaries 4~dev*         [uses yojson]
===== ↻ 2   ↗ 1 =====
Do you want to continue? [Y/n] y

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[yojson.1.6.0] downloaded from cache at https://opam.ocaml.org/cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⊘ removed   kappa-binaries.4~dev
⊘ removed   kappa-library.4~dev
⊘ removed   yojson.1.5.0
∗ installed yojson.1.6.0
[ERROR] The compilation of kappa-library failed at "/home/hmedina/.opam/default/bin/dune build -p kappa-library -j 15".

#=== ERROR while compiling kappa-library.4~dev ================================#
# context     2.0.0 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(git+https://github.com/Kappa-Dev/KaSim.git#49780e1b)
# path        ~/.opam/default/.opam-switch/build/kappa-library.4~dev
# command     ~/.opam/default/bin/dune build -p kappa-library -j 15
# exit-code   1
# env-file    ~/.opam/log/kappa-library-2692-d37749.env
# output-file ~/.opam/log/kappa-library-2692-d37749.out
### output ###
# [...]
# File "dataStructures/setMap.mli", line 236, characters 6-23:
# Error (warning 3): deprecated: Yojson.Basic.json
# json types are being renamed and will be removed in the next Yojson major version. Use type t instead
# File "dataStructures/setMap.mli", line 240, characters 14-31:
# Error (warning 3): deprecated: Yojson.Basic.json
# json types are being renamed and will be removed in the next Yojson major version. Use type t instead
# File "dataStructures/setMap.mli", line 241, characters 17-34:
# Error (warning 3): deprecated: Yojson.Basic.json
# json types are being renamed and will be removed in the next Yojson major version. Use type t instead
# File "dataStructures/setMap.mli", line 242, characters 18-35:
# Error (warning 3): deprecated: Yojson.Basic.json
# json types are being renamed and will be removed in the next Yojson major version. Use type t instead



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build kappa-library 4~dev
└─
┌─ The following changes have been performed (the rest was aborted)
│ ⊘ remove  kappa-binaries 4~dev
│ ⊘ remove  kappa-library  4~dev
│ ↗ upgrade yojson         1.5.0 to 1.6.0
└─

```

**Comment by yarden** [comment]
I'm having an issue installing Kappa suite that's potentially related to this? I tried:

```
$ wget https://github.com/Kappa-Dev/KaSim/archive/v4.1beta.zip
$ cd KaSim-4.1beta
$ make agents
$ make agents
dune build --only-packages kappa-library,kappa-agents
      ocamlc dataStructures/.kappa_generic_toolset.objs/kappa_generic_toolset__JsonUtil.{cmi,cmti} (exit 2)
(cd _build/default && /usr/local/bin/ocamlc.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -w @a -g -bin-annot -I dataStructures/.kappa_generic_toolset.objs -I dataStructures/.kappa_generic_toolset.objs/.private -I /Users/yarden/.opam/system/lib/biniou -I /Users/yarden/.opam/system/lib/easy-format -I /Users/yarden/.opam/system/lib/logs -I /Users/yarden/.opam/system/lib/result -I /Users/yarden/.opam/system/lib/yojson -no-alias-deps -opaque -open Kappa_generic_toolset -o dataStructures/.kappa_generic_toolset.objs/kappa_generic_toolset__JsonUtil.cmi -c -intf dataStructures/jsonUtil.mli)
File "dataStructures/jsonUtil.mli", line 31, characters 25-39:
Error: Unbound type constructor Yojson.Basic.t
       ocaml version/git_version.ml
fatal: Not a git repository (or any of the parent directories): .git
      ocamlc dataStructures/.kappa_generic_toolset.objs/kappa_generic_toolset__SetMap.{cmi,cmti} (exit 2)
(cd _build/default && /usr/local/bin/ocamlc.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -w @a -g -bin-annot -I dataStructures/.kappa_generic_toolset.objs -I dataStructures/.kappa_generic_toolset.objs/.private -I /Users/yarden/.opam/system/lib/biniou -I /Users/yarden/.opam/system/lib/easy-format -I /Users/yarden/.opam/system/lib/logs -I /Users/yarden/.opam/system/lib/result -I /Users/yarden/.opam/system/lib/yojson -no-alias-deps -opaque -open Kappa_generic_toolset -o dataStructures/.kappa_generic_toolset.objs/kappa_generic_toolset__SetMap.cmi -c -intf dataStructures/setMap.mli)
File "dataStructures/setMap.mli", line 234, characters 7-21:
Error: Unbound type constructor Yojson.Basic.t
make: *** [agents] Error 1
```

Any idea what might be causing this? 
Thank you.

**Comment by glucksfall** [comment]
I had problems with dune being incompatible with yojson (see https://opam.ocaml.org/packages/dune/)

here is what I did to compile the most recent KaSim dev version (ubuntu 18.04):
I uninstalled ocaml and opam and reinstalled them from https://github.com/ocaml/ocaml and https://github.com/ocaml/opam. I checkout opam to 2.0 and deleted from .profile the OPAM configuration one-liner (if I could describe as it). I rebooted my laptop and followed compilation instructions from both repositories. Then I installed all dependencies with opam (dune, yojson, lwt, logs, num, re, fmt), then make to compile KaSim 4.1
